include/I_Displayabe.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <SDL2/SDL.h>
+
+// in main add every displayable object to an displayable_array
+// for each object in displayable_array call the display() function
+//      i) DESIGN CHOICE 1
+//      display will return a texture where it will be RenderCopy() inside the loop (in main)
+
+class I_Displayable 
+{
+public:
+    virtual SDL_Texture display(SDL_Renderer *renderer);
```

Comment:
you need to add = 0; to force subclasses to create and define a `display` function.
Search up "Pure Virtual Functions C++".
should be: `virtual SDL_Texture display(SDL_Renderer *renderer) = 0;`